src/lib.rs

```diff
@@ -253,25 +442,30 @@ pub fn authorization_header(
 /// # Examples
 ///
 /// ```
+/// # use oauth_client::DefaultRequestBuilder;
 /// let REQUEST_TOKEN: &str = "http://oauthbin.com/v1/request-token";
 /// let consumer = oauth_client::Token::new("key", "secret");
-/// let bytes = oauth_client::get(REQUEST_TOKEN, &consumer, None, None).unwrap();
-/// let resp = String::from_utf8(bytes).unwrap();
+/// let resp = oauth_client::get::<DefaultRequestBuilder>(REQUEST_TOKEN, &consumer, None, None, &()).unwrap();
 /// ```
-pub fn get(
+pub fn get<RB: RequestBuildah>(
     uri: &str,
     consumer: &Token<'_>,
     token: Option<&Token<'_>>,
     other_param: Option<&ParamList<'_>>,
-) -> Result<Vec<u8>> {
-    let (header, body) = get_header("GET", uri, consumer, token, other_param);
+    client: &RB::ClientBuilder,
```

Comment:
Another possibility: make `get` and `post` associated functions of `RequestBuilder`, and the top-level `get` and `post` functions will be aliases of `DefaultRequestBuilder::get` and `DefaultRequestBuilder::post`.

Reply:
> For example, could we add an associated function client_builder to RequestBuilder that returns a reference to ClientBuilder?

I don't think so. Not without using `static`, since what we pass to `get::<RB>` through the `RB` parameter is just a type it does not hold any state.

> Instead of taking a ClientBuilder as an argument, do we need to take a reference to a type that implements RequestBuilder as an argument?

You mean why by reference instead of by value? I based this signature off of `reqwest::Client` which can be just cloned (holds an Arc internally so it's cheap). I guess both is fine.

> Another possibility: make get and post associated functions of RequestBuilder

That's interesting, you mean DefaultRequestBuilder will no longer need a `fn send` and instead the user will call `DefaultRequestBuilder::get` and `DefaultRequestBuilder::post`? I could try that.

> , and the top-level get and post functions will be aliases of DefaultRequestBuilder::get and DefaultRequestBuilder::post.

Not sure how that would look like exactly. Those `get` and `post` functions would still need to take the exact same arguments, right?

Reply:
I tried to remove this argument, but I couldn't, so I think it's okay to leave it as it is.